mono/mini/interp/interp.c

```diff
@@ -3259,14 +3257,15 @@ interp_exec_method_full (InterpFrame *frame, ThreadContext *context, FrameClause
 			child_frame.stack_args = sp;
 			if (imethod->method->dynamic && csignature->pinvoke) {
 				MonoMarshalSpec **mspecs;
-				MonoMethodPInvoke piinfo;
 				MonoMethod *m;
 
 				/* Pinvoke call is missing the wrapper. See mono_get_native_calli_wrapper */
 				mspecs = g_new0 (MonoMarshalSpec*, csignature->param_count + 1);
-				memset (&piinfo, 0, sizeof (piinfo));
 
-				m = mono_marshal_get_native_func_wrapper (m_class_get_image (imethod->method->klass), csignature, &piinfo, mspecs, code);
+				piinfo = piinfo ? piinfo : g_newa (MonoMethodPInvoke, 1);
```

Comment:
Where is it `free`d?

Reply:
`a` is for `alloca`. When the function returns.

Reply:
https://github.com/mono/mono/blob/a6e6c72c8fc3702db37e1572694a49902f63195f/mono/eglib/glib.h#L292

probably not the best name.

Reply:
Oh, I see.

Reply:
It is subtle I agree.
Should I just spell out `alloca` and the cast?
`g_allocate_stack`? :)

Reply:
It's fine I guess. The name `g_newa` is confusing since it doesn't imply it's `alloca` under the covers.

Reply:
Lets hope this is not called in a loop.

Reply:
@vargaz why would that matter?

Reply:
alloca cannot be called in a loop because it will cause a stack overflow.

Reply:
`piinfo` wouldn't be `NULL` anymore.

Reply:
It is guarded to do the allocation only once so it would not matter here.